src/box/request.c

```diff
@@ -242,7 +242,8 @@ request_handle_sequence(struct request *request, struct space *space,
 		 * code is also invoked on final recovery to restore
 		 * the sequence value from WAL.
 		 */
-		if (likely(mp_read_int64(&key, &value) == 0))
+		if (likely(mp_read_int64(&key, &value) == 0) &&
+		    recovery_state != INITIAL_RECOVERY)
```

Comment:
This looks correct, but I don't understand why you change this code now. Do some tests fail without this check?